src/main/java/org/apache/maven/shared/dependency/analyzer/asm/DependencyClassFileVisitor.java

```diff
@@ -75,6 +75,9 @@ public void visitClass(String className, InputStream in) {
             // some bug inside ASM causes an IOB exception. Log it and move on?
             // this happens when the class isn't valid.
             logger.warn("Unable to process: " + className);
+        } catch (IllegalArgumentException e) {
+            // [MSHARED-1248] should log instead of failing when analyzing a corrupted jar file
+            logger.warn("Unable to process: " + className, e);
```

Comment:
I'm still not sure if it will be the best solution.
Eg. when analyzer will not support jdk in newer version - user will only have a warnings but nothing will be analyzed. 
We know that warnings are many times ignored ...

Reply:
Hi @slawekjaranowski 
I'm not sure how else you propose to solve https://issues.apache.org/jira/browse/MSHARED-1248 
Ideas?

Reply:
As I see all project production and tests classes are analyzed. 
Maybe allowing exclusion for some path, patterns will be what we need.

In specific project we know which classes are broken and which should be excluded.

So we not need to ignore errors when we will can exclude what we want.

@garydgregory - What do you think?